docs(l2): update L2 docs (#5358)
lambdaclass/ethrex: docs/developers/l2/prover.md

After installing the toolchains, a quick test can be performed to check if we have everything installed correctly.

### L1 block proving

ethrex-prover is able to generate execution proofs of Ethereum Mainnet/Testnet blocks. An example binary was created for this purpose in `crates/l2/prover/bench`. Refer to its README for usage.

## Dev Mode

To run the blockchain (`proposer`) and prover in conjunction, start the `Prover`, use the following command:

```sh
make init-prover-<sp1|risc0|exec> # optional: GPU=true
```

### Run the whole system with the prover - In one Machine

> [!NOTE]
> Used for development purposes.

1. `cd crates/l2`
2. `make rm-db-l2 && make down`
   - It will remove any old database, if present, stored in your computer. The absolute path of SQL is defined by [datadir](https://docs.rs/dirs/latest/dirs/fn.datadir.html).
3. `make init`
   - Make sure you have the `solc` compiler installed in your system.
   - Init the L1 in a docker container on port `8545`.
   - Deploy the needed contracts for the L2 on the L1.
   - Start the L2 locally on port `1729`.
4. In a new terminal &rarr; `make init-prover-<sp1|risc0|exec> # GPU=true`.

> After this initialization we should have the prover running in `dev_mode` &rarr; No real proofs.

Reviewer: `sp1` and `risc0` actually generates proofs

> After this initialization we should have the prover running in `dev_mode` &rarr; No real proofs.

Reviewer 2: This is only true, if it was run with `make init-prover-exec`

4. Install [CUDA for Ubuntu](https://developer.nvidia.com/cuda-downloads?target_os=Linux&target_arch=x86_64&Distribution=Ubuntu&target_version=22.04&target_type=deb_local)
   - Install `CUDA Toolkit Installer` first. Then the `nvidia-open` drivers.
5. Reboot
6. Run the following commands:

```sh
sudo apt-get install libssl-dev pkg-config libclang-dev clang
echo 'export PATH=/usr/local/cuda/bin:$PATH' >> ~/.bashrc
echo 'export LD_LIBRARY_PATH=/usr/local/cuda/lib64:$LD_LIBRARY_PATH' >> ~/.bashrc
```

### Run the whole system with a GPU Prover

Two separate machines are recommended for running the `Prover` and the `sequencer` to avoid resource contention. However, for development, you can run them in two separate terminals on the same machine.

- **Machine 1 (or Terminal 1)**: For the `Prover` (GPU is recommended).
- **Machine 2 (or Terminal 2)**: For the `sequencer`/L2 node.

1. **`Prover`/`zkvm` Setup**
   1. `cd ethrex/crates/l2`
   2. You can set the following environment variables to configure the prover:
      - `PROVER_CLIENT_PROVER_SERVER_ENDPOINT`: The address of the server where the client will request the proofs from.

>       - `PROVER_CLIENT_PROVING_TIME_MS`: The amount of time to wait before requesting new data to prove.

Reviewer 2: ```suggestion
      - `PROVER_CLIENT_PROOF_COORDINATOR_URL`: The address of the server where the client will request the proofs from.
      - `PROVER_CLIENT_PROVING_TIME`: The amount of time to wait before requesting new data to prove.
```

      ```env
      # Should be the same as ETHREX_COMMITTER_L1_PRIVATE_KEY and ETHREX_WATCHER_L2_PROPOSER_PRIVATE_KEY
      ETHREX_DEPLOYER_L1_PRIVATE_KEY=<private_key>
      # Should be the same as ETHREX_COMMITTER_L1_PRIVATE_KEY and ETHREX_DEPLOYER_L1_PRIVATE_KEY
      ETHREX_WATCHER_L2_PROPOSER_PRIVATE_KEY=<private_key>
      # Should be the same as ETHREX_WATCHER_L2_PROPOSER_PRIVATE_KEY and ETHREX_DEPLOYER_L1_PRIVATE_KEY
      ETHREX_COMMITTER_L1_PRIVATE_KEY=<private_key>
      # Should be different from ETHREX_COMMITTER_L1_PRIVATE_KEY and ETHREX_WATCHER_L2_PROPOSER_PRIVATE_KEY
      ETHREX_PROOF_COORDINATOR_L1_PRIVATE_KEY=<private_key>
      # Used to handle TCP communication with other servers from any network interface.
      ETHREX_PROOF_COORDINATOR_LISTEN_ADDRESS=0.0.0.0
      # Set to true to randomize the salt.
      ETHREX_DEPLOYER_RANDOMIZE_CONTRACT_DEPLOYMENT=true
      # Set to true if you want SP1 proofs to be required
      ETHREX_L2_SP1=true
      # Check the if the verification contract is present on your preferred network. Don't define this if you want it to be deployed automatically.
      ETHREX_DEPLOYER_SP1_VERIFIER_ADDRESS=<address>

>       # Set to true if you want proofs to be required

Reviewer: ```suggestion
      # Set to true if you want RISC0 proofs to be required
```

>       ETHREX_L2_RISC0=true

Reviewer 3: ```suggestion
      # Set to true if you want RISC0 proofs to be required
      ETHREX_L2_RISC0=true
```
